client/state/data-layer/wpcom-http/actions.js

```diff
@@ -30,15 +40,49 @@ export const http = ( {
 	onFailure,
 	onProgress,
 	...options,
-}, action = null ) => ( {
-	type: WPCOM_HTTP_REQUEST,
-	body,
-	method,
-	path,
-	query: { ...query, apiVersion },
-	formData,
-	onSuccess: onSuccess || action,
-	onFailure: onFailure || action,
-	onProgress: onProgress || action,
-	options,
-} );
+}, action = null ) => {
+	const actionProperties = {
+		onSuccess: onSuccess || action,
+		onFailure: onFailure || action,
+		onProgress: onProgress || action,
+		query: { ...query }
+	};
+
+	// * check parameters *
+
+	// `path` is not optional and it is a string
+	// `method` is not optional and its value can take either 'GET' or 'POST`
+	if (
+		( ! path || typeof path !== 'string' ) ||
+		( ! method || ( method !== 'GET' && method !== 'POST' ) )
+	) {
+		return badRequest();
+	}
+
+	// - One of `apiVersion` of `apiNamespace` should be defined
+	// - `apiVersion` and `apiNamespace` cannot be defined simultaneously
+	if (
+		( typeof apiVersion === 'undefined' && typeof apiNamespace === 'undefined' ) ||
+		( typeof apiVersion !== 'undefined' && typeof apiNamespace !== 'undefined' )
+	) {
+		return badRequest();
+	}
```

Comment:
I think it's fair to gracefully handle malformed requests here, since our callers already have onError handlers. Presumably if we're developing something new we'd notice when it doesn't work.

Additional things to do here would be to add an eslint rule to detect this case so we can't commit the mis-match of version/api namespace, and also update the wpcom.js interface to throw an Error immediately in this case.